client/app/components/queries/EmbedQueryDialog.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Button from 'antd/lib/button';
+import Modal from 'antd/lib/modal';
+import Typography from 'antd/lib/typography';
+import { wrap as wrapDialog, DialogPropType } from '@/components/DialogWrapper';
+import { clientConfig } from '@/services/auth';
+
+const { Text } = Typography;
+
+function EmbedQueryDialog({ dialog, query, visualization }) {
+  const embedUrl = `${clientConfig.basePath}embed/query/${query.id}/visualization/${
+    visualization.id}?api_key=${query.api_key}&${query.getParameters().toUrlParams()}`;
+
+  let snapshotUrl;
+  if (window.snapshotUrlBuilder) {
+    snapshotUrl = window.snapshotUrlBuilder(query, visualization);
+  }
+
+  return (
+    <Modal {...dialog.props} title="Embed Code" footer={(<Button onClick={dialog.dismiss}>Close</Button>)}>
+      <h5 className="m-t-0">IFrame Embed</h5>
+      <div className="m-b-10">
+        <Text data-test="EmbedIframe" copyable code>
+          {`<iframe src="${embedUrl}" width="720" height="391"></iframe>`}
```

Comment:
@gabrieldutra I think it would be better to replace this with `<textarea>` - so user will be able to copy it manually and it will not require ellipsis (I personally use this dialog just to get embed URL when I need to test something, so it's a case for having all the code available).

Reply:
I've considered using a `TextArea` too, but I thought the copy button could be a good feature. I'll check it with the `TextArea` :+1:

Reply:
I agree with @kravets-levko that we shouldn't use an ellipsis here, but why a textarea?